src/hooks/document-comment-hooks.ts

```diff
@@ -119,7 +119,20 @@ export const usePostDocumentComment = (options?: PostDocumentCommentUseMutationO
   return useMutation(postComment, {
     onMutate: async newCommentParams => {
       const { document, comment } = newCommentParams;
-      const queryKey = getCommentsQueryKeyFromMetadata(document);
+      const legacyQueryKey = getCommentsQueryKeyFromMetadata(document);
+      const documentKey = isDocumentMetadata(document) ? document.key : document.path;
```

Comment:
When this is a curriculum document what is documentKey?
I worry that `curriculumPath` will end up with a path that doesn't include something identifying the class or user or something. So then any comments by any user in any class will all be added to the same place. So then comments by students on curriculum will be visible to students in another class. 

But perhaps the comments from other classes are filtered out by a query which includes a class identifier?
If that is true, a code comment describing this would be useful.